library/packages/src/lib/y2packager/license.rb

```diff
@@ -12,40 +12,99 @@
-  # for different languages.
+  # This class represents a license.
   class License
     DEFAULT_LANG = "en_US".freeze
-    # @return [String] License unique identifier
-    attr_reader :id
 
     # @return [Boolean] whether the license has been accepted or not
     attr_reader :accepted
 
     # @return [Hash<String, String>] language -> content
     attr_reader :translations
 
+    # @return [Y2Packager::LicensesFetchers::Base] License fetcher object
+    attr_reader :fetcher
+
     alias_method :accepted?, :accepted
 
+    class << self
+      # Find a license for a given product
+      #
+      # @param product_name [String]    Product's name
+      # @param source       [:rpm,:url] Source to get the license from. For the time being,
+      #   only :rpm is really supported.
+      # @return [License]
+      def find(product_name, source, options = {})
+        return cache[product_name] if cache[product_name]
+
+        # This could be done in the constructor.
+        fetcher = LicensesFetchers.for(source, product_name, options)
+        new_license = License.new(fetcher)
+        return unless new_license.id
+
+        eq_license = cache.values.find { |l| l.id == new_license.id }
+        license = eq_license || new_license
+        cache[product_name] = license
+      end
+
+      def clear_cache
+        @cache = nil
+      end
+
+      private def cache
+        @cache ||= {}
+      end
+    end
+
     # Constructor
     #
     # @param options [Hash<String, String>]
-    def initialize(options = {})
+    def initialize(fetcher)
       @accepted = false
-      @id = id_for(options)
-      @translations = { DEFAULT_LANG => options[:content] }
+      @fetcher = fetcher
+      @translations = {}
+    end
+
+    # License unique identifier
+    #
+    # This identifier is based on the given default language translation.
+    #
+    # @return [String,nil] Unique identifier; nil if the license was not found.
+    def id
+      return @id if @id
+      content = content_for(DEFAULT_LANG)
+      return unless content
+      @id = Digest::MD5.hexdigest(content_for(DEFAULT_LANG))
     end
 
     # Return the license translated content for the given language
     #
+    # It will return the empty string ("") if the license does not exist
+    #
     # @param [String] language
     # @return [String,nil] the license translated content or nil if not found
     def content_for(lang = DEFAULT_LANG)
-      translations[lang]
+      return @translations[lang] if @translations[lang]
+      content = fetcher.license_content(lang)
+      return add_content_for(lang, content) if content
+      content_for(DEFAULT_LANG) unless lang == DEFAULT_LANG
+    end
+
+    # FIXME: Probably the locales should be obtained through the licenses
+    # translations, and probably could be initialized the first time a license
+    # is instantiated.
+    def locales
+      fetcher.license_locales
+    end
+
+    def license_confirmation_required?
```

Comment:
I suggest name `confirmation_required?` license_ prefix is not needed in license class